### DISCUSSION TITLE
Export developer profile

### DISCUSSION THREAD

**Discussion by cwmccann** [fastlane/fastlane#16898] [Q&A]
Hi,

I'm using fastlane to manage my certificates and provisioning profiles for an app.  I've mostly got it working now.  After the application has been built we need to deploy it to a linux server and run tests on it via the WDAServer.  There's a third party tool to run the test from a linux server but it requires a .developerprofile file.  I can generate that file via xcode - export but it's clunky, requires manual intervention, has the risk of exporting more than just the development certificates and probably others.  

Is there a way to create a developer profile from fastlane?  I just need the provisioning profiles and development .p12 files containing both the key and the signing certificate.  

I've tried a bunch of things and I think I can cobble something together by getting the `lane_context[SharedValues::CERT_FILE_PATH]` getting the signing key ids from it, exporting the key from the keychain and merging the two parts together using openssl commands.  I'm guessing that fastlane probably has most of that already in place.

The provisioning profiles are easy to get and put into a zip so it shouldn't be a problem.

**Comment by joshdholtz** [comment]
@cwmccann Hey 👋 This is not something I’ve ever done before and I don’t know if I’ve ever seen a `.developerprofile`. Are you able to link me to any documentation on how that is created in Xcode?

And yup yup, you can definitely create a developer profile and cert with _fastlane_. I would maybe suggest using [match](https://docs.fastlane.tools/actions/match) for this. It will create your profile and cert and store it encrypted in a repository of your choice. You can then pull it down on your other machine and manually decrypt it if you need - https://docs.fastlane.tools/actions/match/#manual-decrypt

**Comment by cwmccann** [comment]
Hi Josh,

Thanks for the response.  The developer profile is created via xcode.  Select preferences, click on the accounts tab.  You'll see all of your apple accounts.  Click on the little cog and select "Export Apple ..."

You'll have to enter a password to protect it.  The resulting file is a zip with all the provisioning profiles, p12 files containing the signing certificates and private keys, along with a plist / keychain with some additional account credentials.  Here's an example:

unzip -l chris.developerprofile
Archive:  chris.developerprofile
DTDKDeveloperProfile closing
  Length      Date    Time    Name
---------  ---------- -----   ----
     3345  00-00-1980 00:00   developer/identities/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.p12
     3199  00-00-1980 00:00   developer/identities/YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY.p12
    20480  00-00-1980 00:00   developer/profiles/aaa-bbb-ccc.mobileprovision
    20052  00-00-1980 00:00   developer/profiles/ddd-eee-fff.mobileprovision
      544  00-00-1980 00:00   developer/accounts.plist
    23952  00-00-1980 00:00   developer/accounts.keychain
---------                     -------
    71572                     6 files 

I'm thinking since I don't need the accounts files, I could write a script to create this file from the match git repository.  Is there a better way using existing fastlane functionality?  I need to do this for multiple apple accounts.

**Reply by joshdholtz** [reply]
@cwmccann Thanks for explaining your process! I had no idea you could export a zip file with these contents from Xcode 😛 I don’t _think_ we have a way to do this exact thing from _fastlane_ at the moment 😔 You could create a local action (http://docs.fastlane.tools/create-action/#local-actions) that pulls down and decrypts `match` and then zips up whatever you need 🤔  That way that logic is contained, you can transfer it between projects, and you can always run it directly from CLI instead of a `Fastfile` by running `fastlane run <action_name>`
